### PR TITLE
ci: use GITHUB_TOKEN instead of PAT

### PR DESCRIPTION
**Pull request by domoritz**
<details>
  <summary>📦 Published PR as canary version: <code>0.97.1--canary.1017.a612074.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install ts-json-schema-generator@0.97.1--canary.1017.a612074.0
  # or 
  yarn add ts-json-schema-generator@0.97.1--canary.1017.a612074.0
  ```
</details>

### CODE REVIEW

**Review by hydrosquall — APPROVED** [review]
As discussed in slack:

> a PAT is needed if the branch that auto commits to is protected . If we leave the branch unprotected, it won't be needed.

https://github.community/t/how-to-push-to-protected-branches-in-a-github-action/16101/10

Since `stable` is unprotected (check: https://a.cl.ly/nOu5Q7L9 ), this should work.

File: .github/workflows/publish-auto.yml [hunk 1]
```diff
@@ -26,7 +26,8 @@ jobs:
         run: yarn --frozen-lockfile
 
       - name: Create release
+        run: npm run release
```
Comment (hydrosquall) [inline .github/workflows/publish-auto.yml:30]: Wondering if using `yarn run release` might have better results than `npm run release`, noticed we are seeing the cleanup error in CI again https://github.com/vega/ts-json-schema-generator/runs/4232722268?check_suite_focus=true

Comment (domoritz) [reply .github/workflows/publish-auto.yml:30]: Let's try in https://github.com/vega/ts-json-schema-generator/pull/1017/commits/f5848fa95efb624c89ec06ab2a207296e1bc211f

Comment (domoritz) [reply .github/workflows/publish-auto.yml:30]: That fails: https://github.com/vega/ts-json-schema-generator/runs/4269550305?check_suite_focus=true

**Pull request merged**